### DISCUSSION TITLE
What's the best way to add @tag directive to generated schema file?

### DISCUSSION THREAD

**Discussion by jwpark511** [MichalLytek/type-graphql#1414] [Q&A] [answered]
Looking for some guidance or an example to add directives to the generated schema file.

```
export default {
  getSchema: (): Promise<GraphQLSchema> =>
    buildSchema({
      resolvers: [
        SomeResolver,
      ],
      emitSchemaFile: path.resolve(__dirname, './schema.graphql'),
    }),
};
```

**Comment by carlocorradini** [comment, marked as answer]
See https://typegraphql.com/docs/emit-schema.html#emit-schema-with-custom-directives

> Currently TypeGraphQL does not directly support emitting the schema with custom directives due to printSchema function limitations from graphql-js.
If we want the custom directives to appear in the generated schema definition file we have to create a custom function that use a third-party printSchema function.
Below there is an example that uses the printSchemaWithDirectives function from [@graphql-tools/utils](https://the-guild.dev/graphql/tools/docs/api/modules/utils_src#printschemawithdirectives):
```ts
import { GraphQLSchema, lexicographicSortSchema } from "graphql";
import { printSchemaWithDirectives } from "@graphql-tools/utils";
import { outputFile } from "type-graphql/dist/helpers/filesystem";

export async function emitSchemaDefinitionWithDirectivesFile(
  schemaFilePath: string,
  schema: GraphQLSchema,
): Promise<void> {
  const schemaFileContent = printSchemaWithDirectives(lexicographicSortSchema(schema));
  await outputFile(schemaFilePath, schemaFileContent);
}
```

> The usage of emitSchemaDefinitionWithDirectivesFile function is the same as with standard emitSchemaDefinitionFile:
```ts
const schema = await buildSchema(/*...*/);

await emitSchemaDefinitionWithDirectivesFile("/path/to/folder/schema.gql", schema);
```

**Reply by carlocorradini** [reply]
@jwpark511 Did you solve your issue?

**Reply by jwpark511** [reply]
Yes, thanks for the docs!
